Sources/Atomic.swift

```diff
@@ -9,37 +9,44 @@
 import Foundation
 
 internal struct _PosixThreadMutex {
-	private var mutex = pthread_mutex_t()
+	private let mutex: UnsafeMutablePointer<pthread_mutex_t>
 
 	init() {
-		let result = pthread_mutex_init(&mutex, nil)
+		mutex = UnsafeMutableRawPointer.allocate(bytes: MemoryLayout<pthread_mutex_t>.size,
+		                                         alignedTo: MemoryLayout<pthread_mutex_t>.alignment)
+		.assumingMemoryBound(to: pthread_mutex_t.self)
+
+		let result = pthread_mutex_init(mutex, nil)
 		precondition(result == 0, "Failed to initialize mutex with error \(result).")
 	}
 
-	mutating func deinitialize() {
-		let result = pthread_mutex_destroy(&mutex)
+	func deinitialize() {
+		let result = pthread_mutex_destroy(mutex)
 		precondition(result == 0, "Failed to destroy mutex with error \(result).")
+
+		UnsafeMutableRawPointer(mutex).deallocate(bytes: MemoryLayout<pthread_mutex_t>.size,
```

Comment:
I think of using a raw pointer here it might make sense to use `mutex.deinitialize(1)` and `mutex.deallocate(1)`? I'm not super familiar with the new pointer APIs though.